# Transformers and BERT exploration

### 0. System and frameworks check.

In [7]:
import sys

print(sys.executable)
print(sys.version)

/Users/dove/Desktop/Projects/clinical-case-intake/.venv/bin/python
3.14.7 (main, Aug  5 2026, 10:29:49) [Clang 21.0.0 (clang-2100.1.1.101)]


In [8]:
import transformers

print(transformers.__version__)

5.17.0


### 1. BERT Tokenizer

In [ ]:
# importing the essential 
from transformers import AutoTokenizer

# telling the transformers lib to load the best tokenizer for pre-trained bert model 
tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")

In [11]:
# starting with some small text 
text = "The patient has hypertension and chest pain."

tokens = tokenizer.tokenize(text)

print(tokens)

['the', 'patient', 'has', 'hyper', '##tension', 'and', 'chest', 'pain', '.']


The word "hypertension" is being splitted because of the BERT technique called WordPiece tokenization. 

Since that word isn't contained into its vocabulary, it divides it in two sub-parts. 

The "##" sign means that the piece who contains it has to be attached to the previous one.

In [13]:
# going towards numerical translation of tokens
encoded = tokenizer(text)

print(encoded)

{'input_ids': [101, 1996, 5776, 2038, 23760, 29048, 1998, 3108, 3255, 1012, 102], 'token_type_ids': [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]}


output:

input_ids:
[101, 1996, 5776, 2038, 23760, 29048, 1998, 3108, 3255, 1012, 102]

token_type_ids:
[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]

attention_mask:
[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]